app/code/Magento/MediaGalleryMetadata/Model/AddExifMetadata.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaGalleryMetadata\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem\DriverInterface;
+use Magento\MediaGalleryMetadata\Model\Jpeg\ReadFile;
+use Magento\MediaGalleryMetadataApi\Api\Data\MetadataInterface;
+use Magento\MediaGalleryMetadataApi\Model\FileInterface;
+use Magento\MediaGalleryMetadataApi\Model\FileInterfaceFactory;
+use Magento\MediaGalleryMetadataApi\Model\SegmentInterface;
+
+/**
+ * Write exif data to the file return updated FileInterface with exif data.
+ */
+class AddExifMetadata
+{
+    private const EXIF_SEGMENT_NAME = 'eXIf';
+
+    private const EXIF_TITLE_SEGMENT = '0x010d';
+
+    private const EXIF_DESCRIPTION_SEGMENT = '0x010e';
+
+    /**
+     * @var DriverInterface
+     */
+    private $driver;
+
+    /**
+     * @var ReadFile
+     */
+    private $fileReader;
+
+    /**
+     * @var FileInterfaceFactory
+     */
+    private $fileFactory;
+
+    /**
+     * @param FileInterfaceFactory $fileFactory
+     * @param DriverInterface $driver
+     * @param ReadFile $fileReader
+     */
+    public function __construct(
+        FileInterfaceFactory $fileFactory,
+        DriverInterface $driver,
+        ReadFile $fileReader
+    ) {
+        $this->fileFactory = $fileFactory;
+        $this->driver = $driver;
+        $this->fileReader = $fileReader;
+    }
+
+    /**
+     * Write exif metadata.
+     *
+     * @param FileInterface $file
+     * @param MetadataInterface $metadata
+     * @param null|SegmentInterface $segment
+     * @return FileInterface
+     */
+    public function execute(FileInterface $file, MetadataInterface $metadata, ?SegmentInterface $segment): FileInterface
+    {
+        if (!is_callable('exif_read_data')) {
+            throw new LocalizedException(
+                __('exif_read_data() must be enabled in php configuration')
+            );
+        }
+
+        $exifData =  $segment ? $segment->getData() : [];
+
+        if ($metadata->getTitle() !== null) {
+            $exifData[self::EXIF_TITLE_SEGMENT][0] = $metadata->getTitle();
+        }
+
+        if ($metadata->getDescription() !== null) {
+            $exifData[self::EXIF_DESCRIPTION_SEGMENT][0] = $metadata->getDescription();
+        }
+
+        $newExifData = '';
+        foreach ($exifData as $tag => $values) {
+            foreach ($values as $value) {
+                $newExifData .= $this->exifMaketag(2, (int) substr($tag, 2), $value);
+            }
+        }
+
+        $this->writeFile($file->getPath(), $newExifData);
```

Comment:
if you are able to update segment data without writing to file it's better to return the content to `WriteExif`. See `\Magento\MediaGalleryMetadata\Model\AddXmpMetadata::execute` for example